src/main/scala/org/apache/spark/deploy/armada/Config.scala

```diff
@@ -68,12 +68,25 @@ private[spark] object Config {
     }
   }
 
-  def transformSelectorsToMap(str: String): Map[String,String] = {
-    if (str.trim.isEmpty) {
-      Map()
-    }
+  /**
+   * Converts a comma separated list of key=value pairs into a Map.
+   * @param str The string to convert.
+   * @return A Map of the key=value pairs.
+   */
+  def commaSeparatedLabelsToMap(str: String): Map[String, String] = {
+    val s = str.trim
+    if (s.isEmpty) Map.empty
     else {
-      str.split(",").map(a => a.split("=")(0) -> a.split("=")(1)).toMap
+      s.split(",").map(_.trim).map { entry =>
+        entry.split("=", 2) match {
+          case Array(key, value) if key.nonEmpty && value.nonEmpty =>
```

Comment:
Here, `value` must be `nonEmpty`, but in `k8sLabelListValidator`, value can be empty.